src/wacks4square/refresh.py

```diff
@@ -0,0 +1,46 @@
+import logging
+from datetime import datetime
+
+from dotenv import load_dotenv
+
+from wacks4square.constants import DATABASE_DIR
+from wacks4square.square import Square
+from wacksbywarby.db import Wackabase
+from wacksbywarby.models import SquareCredentials
+
+load_dotenv()
+
+logger = logging.getLogger("wacks4square")
+
+ACCEPTABLE_DAYS_UNTIL_REFRESH = 14
+
+
+def main(db: Wackabase):
+    logger.info("Checking if token refresh is needed...")
+    creds = db.get_square_creds()
+
+    # check if it is time to refresh
+    expiration_date = creds.expires_at
+    if (expiration_date - datetime.now()).days > ACCEPTABLE_DAYS_UNTIL_REFRESH:
+        return
+
+    logger.info("Time to refresh! (expiration date was %s)", expiration_date)
+    # request a new token from square
+    square = Square(credentials=creds, debug=False)
+    new_token = square.request_token()
+    if not new_token:
+        logger.error("Could not retrieve new token 🙀")
```

Comment:
is there a noisier alert you can add here? can we send a message to discord if this happens?

Reply:
let's make that a separate feature :)